Analyses/src/BFieldPlotter_module.cc

```diff
@@ -0,0 +1,173 @@
+      fhicl::Atom<double>      axisOneMin{Name("axisOneMin"), Comment("Axis one lower edge for plotting (mm) (plane = x/y/z --> axis one = y/x/x)")};
+      fhicl::Atom<double>      axisOneMax{Name("axisOneMax"), Comment("Axis one upper edge for plotting (mm) (plane = x/y/z --> axis one = y/x/x)")};
+      fhicl::Atom<double>      axisTwoMin{Name("axisTwoMin"), Comment("Axis two lower edge for plotting (mm) (plane = x/y/z --> axis two = z/z/y)")};
+      fhicl::Atom<double>      axisTwoMax{Name("axisTwoMax"), Comment("Axis two upper edge for plotting (mm) (plane = x/y/z --> axis two = z/z/y)")};
+      fhicl::Atom<double>      mapBinSize{Name("mapBinSize"), Comment("Map bin size (mm)"), 10.};
+    };
+    typedef art::EDAnalyzer::Table<Config> Parameters;
+
+    explicit BFieldPlotter(const Parameters& pset);
+    virtual ~BFieldPlotter() { }
+
+    void analyze(const art::Event& e);
+
+    void fillHistogram(std::shared_ptr<BFMap> map, art::ServiceHandle<art::TFileService> tfs);
+
+    void endJob();
+
+  private:
+
+    std::string _plane     ; // x, y, or z map plane
+    double      _planeValue; // value to set the plane at (mm)
+    double      _axisOneMin; // sampling axis values (mm)
+    double      _axisOneMax; // sampling axis values (mm)
+    double      _axisTwoMin; // sampling axis values (mm)
+    double      _axisTwoMax; // sampling axis values (mm)
+    double      _mapBinSize; // histogram bin size (mm)
+
+    std::map<std::string,TH2F*> _hMap; //histogram of the map
+    
+  };
+
+  BFieldPlotter::BFieldPlotter(const Parameters& pset) :
+    art::EDAnalyzer(pset)
+    , _plane     (pset().plane     ())
+    , _planeValue(pset().planeValue())
+    , _axisOneMin(pset().axisOneMin())
+    , _axisOneMax(pset().axisOneMax())
+    , _axisTwoMin(pset().axisTwoMin())
+    , _axisTwoMax(pset().axisTwoMax())
+    , _mapBinSize(pset().mapBinSize())
+  {
+    if(_axisOneMin >= _axisOneMax || _axisTwoMin >= _axisTwoMax) {
+      throw cet::exception("BADCONFIG") << "BField mapping plane ill defined: "
+					<< _axisOneMin << " < axisOneValues < " << _axisOneMax << ", "
+					<< _axisTwoMin << " < axisTwoValues < " << _axisTwoMax;
+    }
+    if(_mapBinSize <= 0.) {
+      throw cet::exception("BADCONFIG") << "BField map binning should be >= 0 but given as " << _mapBinSize;
+    }
+
+    if(!(_plane == "x" || _plane == "y" || _plane == "z")) {
+      throw cet::exception("BADCONFIG") << "BField map plane not recognized! Options are x, y, or z but given " 
+					<< _plane.c_str();
+    }
+
+  } //end constructor
+
+  void BFieldPlotter::analyze(const art::Event& event) {
+    art::ServiceHandle<art::TFileService> tfs;
+    // Get the magnetic field manager and the map lists
+    GeomHandle<BFieldManager> bf;
+    const BFieldManager::MapContainerType innerMaps = bf->getInnerMaps();
+    const BFieldManager::MapContainerType outerMaps = bf->getOuterMaps();
+ 
```

Comment:
These should have type:
 const BFieldManager::MapContainerType&

Note the reference.  Not a big deal in this case since your are making a copy of vector of shared pointers to big things, not a copies of big things.  But just do it right every time, even when it does not matter much.

Reply:
Good catch, thank you!